빙고 섞기 기능 P/R

holeman79 opened the pull request:
빙고 섞기 기능 외에 추가적으로 민지님 요청한 수정사항이랑 코드 리팩토링 부분도 적용해서 같이 올립니다~!

- BingoItem 섞기 기능
- 임시 방고 조건에 빙고 아이템이 모두 작성 되어있는지 확인 추가
- updateBingoItem 매개변수에 command 입력한 부분 수정
- BingoMapResponse 내부에서 BingoLineResponse 분리

holeman79 committed:
feat(bingo shuffle) :

- BingoItem 섞기 기능
- 임시 방고 조건에 빙고 아이템이 모두 작성 되어있는지 확인 추가
- updateBingoItem 매개변수에 command 입력한 부분 수정
- BingoMapResponse 내부에서 BingoLineResponse 분리

seongheonsim commented on src/main/kotlin/com/beside/groubing/groubingserver/domain/bingo/domain/BingoBoard.kt:
```diff
@@ -16,7 +15,7 @@ import jakarta.persistence.JoinColumn
 import jakarta.persistence.Table
 import java.time.LocalDate
-
+import kotlin.random.Random
```
@holeman79 
여기 코드 라인이 붙어 있는 것 같은데 한 번 띄워주시면 좋을 것 같습니다!

holeman79 replied:
요고 띄우는 이유는 무엇떄문일까요?

seongheonsim commented on src/main/kotlin/com/beside/groubing/groubingserver/domain/bingo/domain/BingoBoard.kt:
```diff
@@ -48,7 +47,7 @@ class BingoBoard internal constructor(
     @OneToMany(cascade = [CascadeType.ALL])
     @JoinColumn(name = "BINGO_BOARD_ID")
-    val bingoItems: List<BingoItem>
+    var bingoItems: List<BingoItem>
```
@holeman79 
해당 프로퍼티를 val 로 선언하고 MutableList 로 정의해서 리스트 안의 데이터만 변경하는 방법은 어떠세요?

holeman79 replied:
네 불변객체로 처리를 하려고 했는데 사실 빙고섞기는 동시에 발생할 일도 없고 쓸데없이 업데이트 쿼리가 많이 발생하니 성헌님 말대로 val로 선언하고 값만 변경되는게 옳은듯 합니다ㅎㅎ 수정해서 반영할게요!

seongheonsim reviewed (commented):
@holeman79 
기능 개발 하시느라 고생 많으셨습니다👍
간단한 리뷰 남겼으니 시간 괜찮으실 때 확인 부탁드립니다🫡

holeman79 committed:
feat(bingo shuffle) : BingoItem 섞기 기능 수정

- 빙고 섞기 기능 재할당 방식에서 데이터만 변경하는 방법으로 수정.
- P6spy dependency 추가
- Enum 타입 갖는 Entity들에 @Enumerated(EnumType.STRING) 추가

holeman79 committed:
chore(bingo shuffle) : BingoBoard와 import 영역 사이 공백 띄우기
